### PR TITLE
[T4A5][F09-B1]GuoShimin

### PR DESCRIPTION
**Pull request by shimin2016**

### CODE REVIEW

**Comment by shimin2016** [comment]
Ready for review.

File: src/seedu/addressbook/data/AddressBook.java [hunk 1]
```diff
@@ -77,8 +97,12 @@ private void syncTagsWithMasterList(Person person) {
      * @throws DuplicatePersonException if an equivalent person already exists.
      */
     public void addPerson(Person toAdd) throws DuplicatePersonException {
+        UniqueTagList tags = toAdd.getTags();
         syncTagsWithMasterList(toAdd);
         allPersons.add(toAdd);
+        for(Tag t : tags){
+            this.taggings.add(new Tagging(Action.ADD, toAdd, t));
+        }
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/AddressBook.java:105]: Good job for adding this function.

File: src/seedu/addressbook/data/AddressBook.java [hunk 2]
```diff
@@ -111,6 +135,15 @@ public boolean containsTag(Tag key) {
...
         allPersons.remove(toRemove);
+        UniqueTagList tags = toRemove.getTags();
+        for(Tag t : tags){
+            for(Tagging tg : taggings){
+                if(tg.getTag().toString().equals(toRemove.toString())
+                        && tg.getPerson().getName().equals(toRemove.toString())){
+                    tg.changeTagging();
+                }
+            }
+        }
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/AddressBook.java:146]: Good job for adding this function.

File: src/seedu/addressbook/data/AddressBook.java [hunk 3]
```diff
@@ -120,6 +153,13 @@ public void removePerson(ReadOnlyPerson toRemove) throws PersonNotFoundException
      */
     public void removeTag(Tag toRemove) throws TagNotFoundException {
         allTags.remove(toRemove);
+        for(Tagging tg : taggings){
+            if(tg.getTag().toString().equals(toRemove.toString())){
+                tg.changeTagging();
+                break;
+            }
+                
+        }
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/AddressBook.java:162]: Good job for adding this function. But follow our coding convention.

File: src/seedu/addressbook/data/AddressBook.java [hunk 4]
```diff
@@ -143,4 +183,18 @@ public UniquePersonList getAllPersons() {
     public UniqueTagList getAllTags() {
         return new UniqueTagList(allTags);
     }
+    
+    /*
+     * 
+     */
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/AddressBook.java:189]: What is this for? :p

File: src/seedu/addressbook/data/AddressBook.java [hunk 5]
```diff
@@ -143,4 +183,18 @@ public UniquePersonList getAllPersons() {
...
+    private static ArrayList<Tagging> addExistingTags(UniquePersonList people) {
+        ArrayList<Tagging> existing_taggings = new ArrayList<Tagging>();
+        for(Person ppl : people){
+            UniqueTagList temTagList = ppl.getTags();
+            for(Tag t : temTagList){
+                existing_taggings.add(new Tagging(Action.ADD,ppl, t));
+            }
+        }
+        return existing_taggings;
+    }
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/AddressBook.java:199]: Good job for adding this function. But follow our coding convention.

File: src/seedu/addressbook/data/tag/Tagging.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
+package seedu.addressbook.data.tag;
+
+import seedu.addressbook.data.person.Person;
+
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/tag/Tagging.java:4]: You forgot to put the header comment.

**Comment by jeffryhartanto** [comment]
Some comments added. Please close the PR after reading comments. Think where you would print the tagging.

**Pull request closed without merging**